src/main/java/org/openstreetmap/atlas/locale/IsoCountry.java

```diff
@@ -135,6 +159,82 @@ else if (countryCode.length() == ISO3_LENGTH
         return Optional.empty();
     }
 
+    /**
+     * Provides IsoCountry for a valid country display name.
+     *
+     * @param displayCountry
+     *            the display country name, e.g. "United States"
+     * @return an Optional containing the IsoCountry if present
+     */
+    public static Optional<IsoCountry> forDisplayCountry(final String displayCountry)
+    {
+        if (displayCountry != null && DISPLAY_COUNTRY_TO_ISO2.containsKey(displayCountry))
+        {
+            return Optional
+                    .ofNullable(ISO_COUNTRIES.get(DISPLAY_COUNTRY_TO_ISO2.get(displayCountry)));
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Provides IsoCountry for a country display name. If the given display name does not perfectly
+     * match a valid IsoCountry, this will return the closest string match.
+     *
+     * @param displayCountry
+     *            the display country name, e.g. "united stats"
+     * @return an Optional containing the IsoCountry if present
+     */
+    public static Optional<IsoCountry> forDisplayCountryClosestMatch(final String displayCountry)
+    {
+        if (displayCountry != null)
+        {
+            if (DISPLAY_COUNTRY_TO_ISO2.containsKey(displayCountry))
+            {
+                return Optional
+                        .ofNullable(ISO_COUNTRIES.get(DISPLAY_COUNTRY_TO_ISO2.get(displayCountry)));
+            }
+            else
+            {
+                final Optional<String> closestCountry = closestIsoCountry(displayCountry);
+                if (closestCountry.isPresent())
```

Comment:
I recommend logging if we're returning a best approximation to make it fully clear that it wasn't an exact match.